Inc/HALAL/Services/Encoder/Encoder.hpp

```diff
@@ -16,9 +16,9 @@
  */
 class Encoder {
 public:
-	static forward_list<uint8_t> id_manager;
-	static map<pair<Pin, Pin>, TIM_HandleTypeDef*> pin_timer_map;
-	static map<uint8_t, pair<Pin, Pin>> service_ids;
+	static uint8_t id_counter;
+	static map<pair<Pin, Pin>, pair<TIM_HandleTypeDef*, TIM_TypeDef*>> pin_timer_map;
```

Comment:
Ahora que ya está mergeado, podrias usar el TimerPeripheral, quedaria mucho mas limpio
```suggestion
	static map<pair<Pin, Pin>, TimerPeripheral> pin_timer_map;
```